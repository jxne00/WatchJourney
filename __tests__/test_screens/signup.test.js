import React from 'react';
import renderer from 'react-test-renderer';

import SignupScreen from '../../screens/login/Signup';

describe('tests for Sign up screen', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<SignupScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('has 1 child', () => {
    const tree = renderer.create(<SignupScreen />).toJSON();
    expect(tree.children.length).toBe(1);
  });
});
