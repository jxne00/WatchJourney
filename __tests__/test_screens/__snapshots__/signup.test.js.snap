// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for Sign up screen renders correctly 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "#D7DCEA",
      "flex": 1,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      {
        "paddingBottom": 0,
      }
    }
  >
    <RCTScrollView
      style={
        {
          "backgroundColor": "#D7DCEA",
          "paddingBottom": 140,
        }
      }
    >
      <View>
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#D7DCEA",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              {
                "fontFamily": "Poppins-SemiBold",
                "fontSize": 36,
                "marginBottom": "10%",
                "marginTop": "30%",
              }
            }
          >
            Create Account
          </Text>
          <TextInput
            autoCapitalize="none"
            autoCompleteType="off"
            autoCorrect={false}
            autoFocus={true}
            onChangeText={[Function]}
            placeholder="Name"
            placeholderTextColor="gray"
            style={
              {
                "borderBottomColor": "#000",
                "borderBottomWidth": 1,
                "fontFamily": "Poppins-Regular",
                "fontSize": 16,
                "height": 40,
                "marginTop": "6%",
                "width": "80%",
              }
            }
            textContentType="none"
            value=""
          />
          <TextInput
            autoCapitalize="none"
            autoCompleteType="off"
            autoCorrect={false}
            onChangeText={[Function]}
            placeholder="Email"
            placeholderTextColor="gray"
            style={
              {
                "borderBottomColor": "#000",
                "borderBottomWidth": 1,
                "fontFamily": "Poppins-Regular",
                "fontSize": 16,
                "height": 40,
                "marginTop": "6%",
                "width": "80%",
              }
            }
            textContentType="none"
            value=""
          />
          <View
            style={
              {
                "alignItems": "center",
                "borderBottomColor": "#000",
                "borderBottomWidth": 1,
                "flexDirection": "row",
                "height": 40,
                "justifyContent": "space-between",
                "marginTop": "6%",
                "width": "80%",
              }
            }
          >
            <TextInput
              autoCapitalize="none"
              autoCompleteType="off"
              autoCorrect={false}
              onChangeText={[Function]}
              placeholder="Password"
              placeholderTextColor="gray"
              secureTextEntry={true}
              style={
                {
                  "fontFamily": "Poppins-Regular",
                  "fontSize": 16,
                }
              }
              textContentType="password"
              value=""
            />
            <Text />
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "backgroundColor": "#74369D",
                "borderRadius": 10,
                "height": 50,
                "justifyContent": "center",
                "marginTop": "10%",
                "opacity": 1,
                "width": "75%",
              }
            }
          >
            <Text
              style={
                {
                  "color": "#D7DCEA",
                  "fontFamily": "Poppins-SemiBold",
                  "fontSize": 18,
                  "textAlign": "center",
                }
              }
            >
              Sign Up
            </Text>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</RCTSafeAreaView>
`;
